recipe/0001-drop-pyarrow-version-numbers.patch

```diff
@@ -4,70 +4,173 @@ Date: Sat, 30 Apr 2022 10:16:22 -0400
++    raise
+     _ABLE_TO_COMPILE_EXTENSIONS = False
+
+ if _ABLE_TO_COMPILE_EXTENSIONS:
+@@ -72,16 +73,16 @@ if _ABLE_TO_COMPILE_EXTENSIONS:
          # upgraded
          arrow_libs_to_copy = {
--            "linux": ["libarrow.so.800", "libarrow_python.so.800", "libparquet.so.800"],
-+            "linux": ["libarrow.so", "libarrow_python.so", "libparquet.so"],
+             "linux": [
+-                "libarrow.so.1000",
+-                "libarrow_dataset.so.1000",
+-                "libarrow_python.so.1000",
+-                "libparquet.so.1000",
++                "libarrow.so",
++                "libarrow_dataset.so",
++                "libarrow_python.so",
++                "libparquet.so",
+             ],
              "darwin": [
--                "libarrow.800.dylib",
--                "libarrow_python.800.dylib",
--                "libparquet.800.dylib",
+-                "libarrow.1000.dylib",
+-                "libarrow_dataset.1000.dylib",
+-                "libarrow_python.1000.dylib",
+-                "libparquet.1000.dylib",
 +                "libarrow.dylib",
++                "libarrow_dataset.dylib",
 +                "libarrow_python.dylib",
 +                "libparquet.dylib",
              ],
-             "win32": ["arrow.dll", "arrow_python.dll", "parquet.dll"],
-         }
- 
+             "win32": [
+                 "arrow.dll",
+@@ -93,16 +94,16 @@ if _ABLE_TO_COMPILE_EXTENSIONS:
+
          arrow_libs_to_link = {
--            "linux": ["libarrow.so.800", "libarrow_python.so.800", "libparquet.so.800"],
-+            "linux": ["libarrow.so", "libarrow_python.so", "libparquet.so"],
+             "linux": [
+-                "libarrow.so.1000",
+-                "libarrow_dataset.so.1000",
+-                "libarrow_python.so.1000",
+-                "libparquet.so.1000",
++                "libarrow.so",
++                "libarrow_dataset.so",
++                "libarrow_python.so",
++                "libparquet.so",
+             ],
              "darwin": [
--                "libarrow.800.dylib",
--                "libarrow_python.800.dylib",
--                "libparquet.800.dylib",
+-                "libarrow.1000.dylib",
+-                "libarrow_dataset.1000.dylib",
+-                "libarrow_python.1000.dylib",
+-                "libparquet.1000.dylib",
 +                "libarrow.dylib",
++                "libarrow_dataset.dylib",
 +                "libarrow_python.dylib",
 +                "libparquet.dylib",
              ],
-             "win32": ["arrow.lib", "arrow_python.lib", "parquet.lib"],
-         }
--- 
-2.37.2
+             "win32": [
+                 "arrow.lib",
+@@ -147,16 +148,20 @@ if _ABLE_TO_COMPILE_EXTENSIONS:
+                 ext.include_dirs.append(ARROW_ITERATOR_SRC_DIR)
+                 ext.include_dirs.append(LOGGING_SRC_DIR)
+
++                std_version = "c++17"
++                if pyarrow_version[0] < 10:
++                    std_version = "c++11"
```

Comment:
This is wrong in conda-forge. Everything is built with C++17

Reply:
I'll try to drop this after adjusting the rpaths. I think I was getting compiler errors for older arrow versions which prompted me to mirror how the upstream lib configured their build before they upgraded to arrow 10.